client/api/src/cht.rs

```diff
@@ -150,7 +150,7 @@ pub fn check_proof<Header, Hasher>(
 			.map(|mut map| map
 				.remove(local_cht_key)
 				.expect("checked proof of local_cht_key; qed"))
-			.map_err(|e| ClientError::from(e)),
+			.map_err(|e| ClientError::from_state(e)),
```

Comment:
```suggestion
			.map_err(ClientError::from_state),
```

Reply:
Does unfortunately not work :) tried that first, but with the trait object you need an exact match with the arg, otherwise latest stable can not resolve the `fn` to use.

Reply:
Actually, not sure, this might have overlayed with the issue of explicit `Send + Sync` trait bounds being required.